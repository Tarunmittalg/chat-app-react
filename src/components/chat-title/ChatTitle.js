import React from 'react';

import TrashIcon from '../controls/icons/trash-icon/TrashIcon';

import './ChatTitle.scss';

const ChatTitle = ({ selectedConversation, onDeleteConversation }) => {
    let chatTitleContents = null;

    if (selectedConversation) {
        chatTitleContents = (
            <>
                <span>{ selectedConversation.title }</span>
                <div onClick={ () => { onDeleteConversation(); } } title="Delete Conversation">
                    <TrashIcon />
                </div>
            </>
        );
    }

    return (
        <div id="chat-title">
            { chatTitleContents }
        </div>
    );
}

export default ChatTitle;