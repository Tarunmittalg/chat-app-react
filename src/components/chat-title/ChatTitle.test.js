import React from 'react';
import renderer from 'react-test-renderer';

import ChatTitle from './ChatTitle';

test('Chat title loads with correct title for selected conversation', () => {
    const selectedConversation = { title: 'My Conversation' };
    
    const component = renderer.create(
        <>
            <ChatTitle selectedConversation={selectedConversation} />
        </>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});

test('contents missing when there is no selected conversation', () => {
    const component = renderer.create(
        <>
            <ChatTitle />
        </>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});
