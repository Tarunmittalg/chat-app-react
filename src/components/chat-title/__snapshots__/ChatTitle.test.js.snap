// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat title loads with correct title 1`] = `
<div
  id="chat-title"
>
  <span>
    My Conversation
  </span>
  <img
    alt="Delete Conversation"
    onClick={[Function]}
    src={
      Object {
        "ReactComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "__esModule": true,
        "default": "trash-logo.svg",
      }
    }
  />
</div>
`;
